defs.bzl

```diff
@@ -62,6 +80,7 @@ shader_binary = rule(
             default = '',
             values = [''] + _shader_stages,
         ),
+        "opts": attr.string_list(default = []),
```

Comment:
instead of a general `opts` could we emulate what the `cc_*` libraries do and have a `defines` attribute for defines etc? https://bazel.build/reference/be/c-cpp#cc_library.defines